pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +436,40 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig) (*vm.VirtualMachine, error) {
+	defer trace.End(trace.Begin(""))
+
+	var vms []mo.VirtualMachine
+
+	// Retrieve config.hardware property of all machines under this view
+	err := m.view.Retrieve(op, []string{"VirtualMachine"}, []string{"config.hardware", "runtime.powerState"}, &vms)
+	if err != nil {
+		return nil, err
+	}
+
+	// iterate over them to see whether they have the disk we want
+	for i := range vms {
+		name := vms[i].Name
+		state := vms[i].Runtime.PowerState
+		if state != types.VirtualMachinePowerStatePoweredOn && state != types.VirtualMachinePowerStateSuspended {
```

Comment:
I manually tested to attach one disk in one suspended vm, that works well, and if I try to start the suspended vm after attached it's vmdk, the vm start failed.
So I'm wondering why suspended vm is not filtered out?